WalletWasabi/WabiSabi/Client/DependencyGraphTaskScheduler.cs

```diff
@@ -142,14 +143,17 @@ public async Task StartReissuancesAsync(IEnumerable<AliceClient> aliceClients, B
 			}
 		}
 
-		public async Task StartOutputRegistrationsAsync(IEnumerable<TxOut> txOuts, BobClient bobClient, CancellationToken cancellationToken)
+		public async Task StartOutputRegistrationsAsync(IEnumerable<TxOut> txOuts, BobClient bobClient, TimeSpan remainingTimeForOutputRegistration, CancellationToken cancellationToken)
 		{
 			List<Task> outputTasks = new();
 
 			using CancellationTokenSource ctsOnError = new();
 			using CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, ctsOnError.Token);
 
-			foreach (var (node, txOut) in Enumerable.Zip(Graph.Outputs, txOuts))
+			// Gets the list of scheduled dates/time in the remaining available time frame when each alice has to be registered.
+			var scheduledDates = remainingTimeForOutputRegistration.SamplePoisson(txOuts.Count());
```

Comment:
Why does `SamplePoisson` contain `.OrderBy(t => t)`? But I guess `txOuts` are in some random order so it may not be an issue.. or?

cc @nothingmuch

Reply:
A Poisson distribution is defined by exponentially distributed intervals between successive events, or equivalently that events are distributed uniformly on the time axis. We're sampling absolute times uniformly within a time window, so order has no effect on the times themselves, as they are implicitly ordered when the delays are in place.

However, it does affect which output registration is assigned which delay. This order is set by the decomposition, and the dependency graph `Outputs` property, which is used by `DependencyGraphScheduler`, preserves it.

In the future this will be shuffled, and I think the randomized decomposer also produces a random ordering so it should be shuffled right now, but I did not verify.

As an optimization, amounts deemed more likely to be registered by other users could be biased to be registered earlier, but exactly how to do this requires more careful study to ensure this doesn't leak output-output link information, which if I'm not mistaken should be possible if we sample each delay independently but parameterized by the amount's predicted likelyhood (all clients must compute the denomination likelyhood the same way).